chanacceptor/rpcacceptor.go

```diff
@@ -0,0 +1,198 @@
+package chanacceptor
+
+import (
+	"errors"
+	"sync"
+	"sync/atomic"
+	"time"
+)
+
+const (
+	// DefaultRPCTimeout is the default time that the Accept method will
+	// listen for responses from the RPC server, after which it will timeout.
+	DefaultRPCTimeout = 15 * time.Second
+)
+
+// ErrNoChainedAcceptor is returned when the RPCAcceptor is unable to retrieve
+// the ChainedAcceptor singleton.
+var ErrNoChainedAcceptor = errors.New("unable to get chained acceptor")
+
+// OpenChannelResponse is a struct containing the response from the RPC client
+// which includes whether or not they accept the open channel request and to
+// which PendingChannelID the response is to.
+type OpenChannelResponse struct {
+	// Accept is the decision from the RPC client that determines whether or not
+	// the client wishes to accept or reject the OpenChannel request.
+	Accept bool
+
+	// PendingChannelID represents the temporary id assigned to channels before
+	// they have a permanent channel id. This is used to determine which
+	// potential channel the above Accept applies to.
+	PendingChannelID [32]byte
+}
+
+// Client represents an RPC streaming client that responds to OpenChannel queries
+// issued by the RPCAcceptor.
+type Client struct {
+	// RequestChan sends the *OpenChannelRequest to the RPC client.
+	RequestChan chan *OpenChannelRequest
+
+	// ResponseChan receives OpenChannelResponse from the RPC client.
+	ResponseChan chan OpenChannelResponse
+}
+
+// RPCAcceptor represents the RPC-controlled variant of the ChannelAcceptor.
+// One RPCAcceptor allows one RPC client represented by the Client struct.
+type RPCAcceptor struct {
+	started int32 // To be used atomically.
+	stopped int32 // To be used atomically.
+
+	Client *Client
+
+	acceptChan       chan *OpenChannelRequest
+	acceptClientsMtx sync.RWMutex
+	acceptClients    map[[32]byte]chan bool
+
+	id uint64
+
+	quit chan struct{}
+	wg   sync.WaitGroup
+}
+
+// Accept is a predicate on the OpenChannelRequest which is sent to the RPC
+// client who will respond with the ultimate decision. This assumes a registered
+// client exists.
+func (r *RPCAcceptor) Accept(req *OpenChannelRequest) bool {
+
+	// Register a new channel for the PendingChannelID associated with this
+	// OpenChannelRequest.
+	responseChan := make(chan bool)
+	r.acceptClientsMtx.Lock()
+	r.acceptClients[req.OpenChanMsg.PendingChannelID] = responseChan
+	r.acceptClientsMtx.Unlock()
+
+	select {
+	case r.acceptChan <- req:
```

Comment:
I find this way of communicating with the RPCserver a bit convulated. 

Not totally sure what would be a better way thought. Would it work having the rpcserver define a closure that would do the demultiplexing? 

```go



NewAcceptor(
    func(req *OpenChannelRequest) bool {
             respChan := make chan
             newRequets <- {req, respChan}

             return <-respChan
    },
)

go func() {
     for {
         resp := stream.Recv()
          resps <- resp
      }
} 

requests := make(map[pendingID]chan response)
for {
     select{
     case req := <-newRequests:
              requests[pendingid] = respChan
             stream.Send(req)
     case resp := <-resps:
              respChan, ok := requests[pendindid]
              if !ok {continue}

             respChan <- resp
             delete(requets, pendingID)
     }

}
```

Reply:
This would also let us remove `Start` & `Stop` in the `RPCAcceptor`